includes/services/wsl.authentication.php

```diff
@@ -562,23 +500,78 @@ function wsl_process_login_get_user_data( $provider, $redirect_to )
 	$user_id = (int) wsl_get_stored_hybridauth_user_id_by_provider_and_provider_uid( $provider, $hybridauth_user_profile->identifier );
 
 	// if not found in wslusersprofiles, then check his verified email
-	if( ! $user_id && ! empty( $hybridauth_user_profile->emailVerified ) )
+	if( ! $user_id && ! empty( $hybridauth_user_email_verified ) )
 	{
 		// check if the verified email exist in wp_users
-		$user_id = (int) wsl_wp_email_exists( $hybridauth_user_profile->email );
+		$user_id = (int) wsl_wp_email_exists( $hybridauth_user_email_verified );
 
 		// the user exists in Wordpress
 		$wordpress_user_id = $user_id;
 
 		// check if the verified email exist in wslusersprofiles
 		if( ! $user_id )
 		{
-			$user_id = (int) wsl_get_stored_hybridauth_user_id_by_email_verified( $hybridauth_user_profile->email );
+			$user_id = (int) wsl_get_stored_hybridauth_user_id_by_email_verified( $hybridauth_user_email_verified );
 		}
 	}
 
 	/* 4 Deletegate detection of user id to custom filters hooks */
 
+	// HOOKABLE:
+	$user_id = apply_filters( 'wsl_hook_process_login_alter_user_id', $user_id, $provider, $hybridauth_user_profile );
```

Comment:
This PR adds back the `wsl_hook_process_login_alter_user_id` which will solve #175 too.